daml-script/test/src/com/digitalasset/daml/lf/engine/script/test/TlsIT.scala

```diff
@@ -10,47 +10,99 @@ import com.daml.ledger.api.testing.utils.SuiteResourceManagementAroundAll
 import com.daml.ledger.api.tls.TlsConfiguration
 import com.daml.lf.data.Ref._
 import com.daml.lf.engine.script.ScriptTimeMode
+import com.daml.lf.speedy.SValue
 import org.scalatest._
 
+import scala.concurrent.Future
+
 final class TlsIT
     extends AsyncWordSpec
     with SandboxParticipantFixture
     with Matchers
-    with SuiteResourceManagementAroundAll {
+    with SuiteResourceManagementAroundAll
+    with OCSPResponderFixture {
+
   val (dar, envIface) = readDar(stableDarFile)
 
-  val List(serverCrt, serverPem, caCrt, clientCrt, clientPem) = {
-    List("server.crt", "server.pem", "ca.crt", "client.crt", "client.pem").map { src =>
+  val List(
+    serverCrt,
+    serverPem,
+    caCrt,
+    clientCrt,
+    clientPem,
+    index,
+    ocspKey,
+    ocspCert,
+    clientRevokedCrt,
+    clientRevokedPem) = {
+    List(
+      "server.crt",
+      "server.pem",
+      "ca.crt",
+      "client.crt",
+      "client.pem",
+      "index.txt",
+      "ocsp.key.pem",
+      "ocsp.crt",
+      "client-revoked.crt",
+      "client-revoked.pem").map { src =>
       Some(new File(rlocation("ledger/test-common/test-certificates/" + src)))
     }
   }
 
+  val indexPath = index.get.getAbsolutePath
+  val caCertPath = caCrt.get.getAbsolutePath
+  val ocspKeyPath = ocspKey.get.getAbsolutePath
+  val ocspCertPath = ocspCert.get.getAbsolutePath
+  val clientCertPath = clientCrt.get.getAbsolutePath
+
   override def timeMode = ScriptTimeMode.WallClock
 
   override protected def config =
     super.config
-      .copy(tlsConfig = Some(TlsConfiguration(enabled = true, serverCrt, serverPem, caCrt)))
-
-  "DAML Script against ledger with TLS" can {
-    "test0" should {
-      "create and accept Proposal" in {
-        for {
-          clients <- participantClients(
-            tlsConfiguration = TlsConfiguration(
-              enabled = true,
-              keyCertChainFile = clientCrt,
-              keyFile = clientPem,
-              trustCertCollectionFile = caCrt
-            ))
-          _ <- run(
-            clients,
-            QualifiedName.assertFromString("ScriptTest:test0"),
-            dar = dar,
-          )
-        } yield
-        // No assertion, we just want to see that it succeeds
-        succeed
-      }
+      .copy(
+        tlsConfig = Some(
+          TlsConfiguration(enabled = true, serverCrt, serverPem, caCrt, revocationChecks = true)))
+
+  "The test runner" should {
+    "be configured to enable OCSP revocation checks" in {
+      sys.props.get("com.sun.net.ssl.checkRevocation") shouldBe Some("true")
     }
   }
+
+  "DAML Script against ledger with TLS" should {
+    "create and accept Proposal" in {
+      executeSampleRequest(clientCrt, clientPem)
+        .map(_ => succeed) // No assertion, we just want to see that it succeeds
+    }
+
+    "fail to create and accept Proposal with a revoked client certificate" in {
```

Comment:
This seems a bit of an odd place to test this. You are testing generic ledger functionality which is not specific to DAML Script in any way. The TLS tests here are really supposed to test that the client configuration in DAML Script works correctly and we can connect to the ledger. Not quite sure which of the various ledger test suites is the correct one here. @SamirTalwar-DA can probably help you.

Reply:
That's a good point; we could test this at a much lower level. I expect this can be a test using `SandboxNextFixture` in _ledger/sandbox_.

Reply:
I've moved tests to the `ledger/sandbox` package.